spec/service_providers/pact_ruby_fetch_pacts_test.rb

```diff
@@ -0,0 +1,292 @@
+require 'pact/pact_broker/fetch_pacts'
+
+Pact.service_consumer "Pact Ruby" do
+  has_pact_with "Pact Broker" do
+    mock_service :pact_broker do
+      port 1234
+      pact_specification_version "2.0.0"
+    end
+  end
+end
+
+describe Pact::PactBroker::FetchPacts, pact: true do
+
+  let(:get_headers) {{:Accept => 'application/hal+json'}}
+
+  describe "fetch pacts" do
+    let(:provider) {'provider-name'}
+    let(:broker_base_url) {'http://localhost:1234/'}
+    let(:basic_auth_options) {{username: 'foo', password: 'bar'}}
+
+    before do
+      pact_broker
+        .given('all the relations exists in the index resource')
+        .upon_receiving('a request for the index resource')
+        .with(
+          method: :get,
+          path: '/',
+          headers: get_headers
+        ).
+        will_respond_with(
+          status: 200,
+          headers: {},
+          body: {
+            _links: {
+              self: {
+                href: broker_base_url
+              },
+              'pb:latest-provider-pacts' => {
+                href: 'http://localhost:1234/pacts/provider/{provider}/latest',
+                title: 'Latest pacts by provider'
+              },
+              'pb:latest-provider-pacts-with-tag' => {
+                'href' => 'http://localhost:1234/pacts/provider/{provider}/latest/{tag}',
+                'title' => 'Latest pacts by provider with the specified tag'
+              },
+              :'pb:provider-pacts-with-tag' => {
+                'href' => 'http://localhost:1234/pacts/provider/{provider}/tag/{tag}',
+                'title' => 'All pact versions for the provider with the specified consumer version tag'
+              },
+              'pb:provider-pacts' => {
+                'href' => 'http://localhost:1234/pacts/provider/{provider}',
+                'title' => 'All pact versions for the specified provider'
+              }
+            }
+          }
+        )
+    end
+
+    context 'retrieving latest pacts by provider' do
+      let(:tags) {nil}
+      let(:all_pacts) {false}
+
+      before do
+        pact_broker
+          .given('pacts exist between consumer and provider')
+          .upon_receiving('a request to retrieve the latest pacts for provider')
+          .with(
+            method: :get,
+            path: '/pacts/provider/provider-name/latest',
+            headers: get_headers
+          ).
+          will_respond_with(
+            status: 200,
+            headers: {},
+            body: {
+              _links: {
+                self: {
+                  href: 'http://localhost:1234/pacts/provider/provider-name/latest',
+                  title: 'Latest pact versions for the provider'},
+                provider: {
+                  href: 'http://localhost:1234/pacticipants/provider-name',
+                  'title': 'provider-name'
+                },
+                pacts: [
+                  {'href': 'pact-brker-url-for-consumer-1'
```

Comment:
I see you're using exact string matching here. What do you think is going to happen when this href is verified?